packages/server/src/worker/processJobSyncAlgolia.ts

```diff
@@ -29,36 +29,59 @@ export async function processJobSyncAlgolia(
   job.updateProgress(25);
 
   let wasUpdated = false;
-  Resource.getAll()
+  let errorMessage = null;
+  Resource.find({})
     .then(resourceDocuments =>
       Promise.all(resourceDocuments.map(resourceDocumentToResource))
     )
+    .catch(error => {
+      throw error;
+    })
     .then(resources => {
       job.updateProgress(50);
       console.info(
         `${job.name}[${job.id}]\t: Resources have been successfully retrieved from the database. Preparing ${resources.length} resources as Algolia Records`
       );
 
-      const updatedAlgoliaIndex = resources.map(
-        ({ resourceId, name, description, subcategories }): TAlgoliaHit => ({
-          objectID: resourceId,
-          name,
-          description,
-          subcategories: subcategories.map(s => s.name).join(","),
-        })
+      const updatedAlgoliaIndex = resources.reduce(
+        (
+          container,
+          { resourceId, name, description, subcategories, deleted }
+        ) => {
+          const resourceIndex = {
+            objectID: resourceId,
+            name,
+            description,
+            subcategories: subcategories.map(s => s.name).join(","),
+          };
+
+          deleted
+            ? container.delete.push(resourceIndex.objectID)
+            : container.update.push(resourceIndex);
+
+          return container;
```

Comment:
```suggestion
return deleted ?
   { update: container.update, deleted: [...container.delete, resourceIndex.ObjectId] }
   :
   { update: [...container.update, resourceIndex], deleted: container.deleted };
```

Reply:
functionally these do the same thing, but not mutating objects is just about always a good idea.